scalalib/src/mill/scalalib/TestModuleUtil.scala

```diff
@@ -136,6 +96,332 @@ private[scalalib] object TestModuleUtil {
       }
     if (selectors.nonEmpty && filteredClassLists.isEmpty) throw doesNotMatchError
 
+    val result = if (useTestStealingScheduler) {
+      runTestStealingScheduler(
+        useArgsFile,
+        testFramework,
+        runClasspath,
+        testClasspath,
+        args,
+        testrunnerEntrypointClasspath,
+        forkEnv,
+        testSandboxWorkingDir,
+        forkWorkingDir,
+        javaHome,
+        props,
+        jvmArgs,
+        testRunnerClasspathArg,
+        resourceEnv,
+        filteredClassLists
+      )
+    } else {
+      runTestDefault(
+        useArgsFile,
+        testFramework,
+        runClasspath,
+        testClasspath,
+        args,
+        testrunnerEntrypointClasspath,
+        forkEnv,
+        testSandboxWorkingDir,
+        forkWorkingDir,
+        javaHome,
+        props,
+        jvmArgs,
+        testRunnerClasspathArg,
+        resourceEnv,
+        filteredClassLists
+      )
+    }
+
+    result match {
+      case Result.Failure(errMsg) => Result.Failure(errMsg)
+      case Result.Success((doneMsg, results)) =>
+        if (results.isEmpty && selectors.nonEmpty) throw doesNotMatchError
+        try handleResults(doneMsg, results, Task.ctx(), testReportXml)
+        catch {
+          case e: Throwable => Result.Failure("Test reporting failed: " + e)
+        }
+    }
+  }
+
+  private def loadArgsAndProps(useArgsFile: Boolean, forkArgs: Seq[String]) = {
+    if (useArgsFile) {
+      val (props, jvmArgs) = forkArgs.partition(_.startsWith("-D"))
+      val sysProps =
+        props
+          .map(_.drop(2).split("[=]", 2))
+          .map {
+            case Array(k, v) => k -> v
+            case Array(k) => k -> ""
+          }
+          .toMap
+
+      jvmArgs -> sysProps
+    } else {
+      forkArgs -> Map()
+    }
+  }
+
+  private def callTestRunnerSubprocess(
+      useArgsFile: Boolean,
+      testFramework: String,
+      runClasspath: Seq[PathRef],
+      testClasspath: Seq[PathRef],
+      args: Seq[String],
+      testrunnerEntrypointClasspath: Seq[PathRef],
+      forkEnv: Map[String, String],
+      testSandboxWorkingDir: Boolean,
+      forkWorkingDir: os.Path,
+      javaHome: Option[os.Path],
+      props: Map[String, String],
+      jvmArgs: Seq[String],
+      testRunnerClasspathArg: String,
+      resourceEnv: Map[String, String],
+      baseFolder: os.Path,
+      selector: Either[Seq[String], os.Path]
```

Comment:
I think these parameter lists are getting copy-pasty enough that it's time to create a `class TestModuleUtil` that takes all these in the constructor and then has methods than use the various configuration values without needing to tediously pass around the whole list every defsite and callsite.